pkg/gobinary/parse.go

```diff
@@ -29,14 +29,29 @@ func Parse(r io.Reader) ([]types.Library, error) {
 	for scanner.Scan() {
 		line := scanner.Text()
 		ss := strings.Fields(line)
-		if len(ss) != 4 {
+
+		// Since we only use "dep" and "=>" which are both at least 3 column, skip if
+		// length is shorter.
+		if len(ss) < 3 {
 			continue
 		}
 
-		libs = append(libs, types.Library{
-			Name:    ss[1],
-			Version: ss[2],
-		})
+		switch ss[0] {
+		case "dep":
+			libs = append(libs, types.Library{
+				Name:    ss[1],
+				Version: ss[2],
+			})
+		case "=>":
+			// replace replaces the previous entry
+			prev := len(libs) - 1
```

Comment:
IMHO.

It is not possible by golang elf design, but len​​(libs) may return 0.

Reply:
Should i return an error or skip that line only or assume this is not a valid go binary and `return nil, nil`?

Reply:
I prefer returning an error here.